src/base/net/downloadhandlerimpl.cpp

```diff
@@ -42,7 +42,7 @@ namespace
 {
     bool saveToFile(const QByteArray &replyData, QString &filePath)
     {
-        QTemporaryFile tmpfile {Utils::Fs::tempPath() + "XXXXXX"};
+        QTemporaryFile tmpfile {filePath.isEmpty() ? Utils::Fs::tempPath() + "XXXXXX" : filePath};
```

Comment:
Do you really need to use QTemporaryFile in case when user provides destination file path?

Reply:
keeps code simple.

Reply:
>keeps code simple.

Doesn't it make the logic wrong?
The correct one should be as the following:
>if saveToFile is set, the file is saved in destFileName

But QTemporaryFile will use another (transformed) file name if the destination file already exists for some reason.